gpgwrapper/gpgwrapper_test.go

```diff
@@ -99,6 +101,198 @@ func TestImportPublicKey(t *testing.T) {
+			"A999 B749 8D1A 8DC4 73E5  3C92 309F 635D AD1B 5517",
+			false,
+		},
+		{
+			"a999b7498d1a8dc473e53c92309f635dad1b5517",
+			"A999 B749 8D1A 8DC4 73E5  3C92 309F 635D AD1B 5517",
+			false,
+		},
+		{
+			"DEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEEFD",
+			"",
+			true, // error: too long
+		},
+		{
+			"DEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEE",
+			"",
+			true, // error: too long
+		},
+		{
+			"DEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEEFG",
+			"",
+			true, // error, contains bad character G
+		},
+		{
+			"",
+			"",
+			true, // error, empty
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("parseFingerprint(%v)", test.inputString), func(t *testing.T) {
+			gotOutput, err := parseFingerprint(test.inputString)
+
+			var gotError bool = err != nil
+
+			if gotError != test.shouldReturnError {
+				t.Errorf("expected shouldReturnError=%v, got err=%v", test.shouldReturnError, err)
+			}
+
+			if test.expectedOutput != gotOutput {
+				t.Errorf("expected output='%s', got='%s'", test.expectedOutput, gotOutput)
+			}
+
+		})
+	}
+}
+
+func TestParseTimestamp(t *testing.T) {
+	ignore := time.Now()
+
+	var timestampTests = []struct {
+		inputString       string
+		expectedTime      time.Time
+		shouldReturnError bool
+	}{
+		{
+			"1536317435", // in BST
+			time.Date(2018, 9, 7, 10, 50, 35, 0, time.UTC),
+			false,
+		},
+		{
+			"1516034100", // in GMT
+			time.Date(2018, 1, 15, 16, 35, 0, 0, time.UTC),
+			false,
+		},
+		{
+			"-10",
+			time.Date(1969, 12, 31, 23, 59, 50, 0, time.UTC),
+			false,
+		},
+		{
+			"1516034100a", // bad: not an int
+			ignore,
+			true, // should return error
+		},
+	}
+
+	for _, test := range timestampTests {
+		t.Run(fmt.Sprintf("parseTimestamp(%v)", test.inputString), func(t *testing.T) {
+			gotTime, err := parseTimestamp(test.inputString)
+
+			var gotError bool = (err != nil)
+
+			if (gotError && !test.shouldReturnError) || (!gotError && test.shouldReturnError) {
+				t.Fatalf("expected shouldReturnError=%v, got err=%v", test.shouldReturnError, err)
+			}
+
+			if gotError {
+				if gotTime != nil {
+					t.Fatalf("returned an error, expected *time.Time to be nil but it wasn't. err: %v, time: %v", err, gotTime)
+				}
+
+			} else {
+				if gotTime == nil {
+					t.Fatalf("got nil *time.Time but err wasn't set")
+				}
+
+				if test.expectedTime != *gotTime {
+					t.Errorf("expected time: %v, got: %v", test.expectedTime, gotTime)
+				}
+			}
+
+		})
+	}
+}
+
+func assertEqual(t *testing.T, expected SecretKeyListing, got SecretKeyListing) {
+	t.Helper()
+	if expected.fingerprint != got.fingerprint {
+		t.Fatalf("fingerprints don't match, expected '%s', got '%s'",
+			expected.fingerprint, got.fingerprint)
+	}
+
+	if expected.created != got.created {
+		t.Fatalf("created don't match, expected '%s', got '%s'", expected.created, got.created)
+	}
+
+	if len(expected.uids) != len(got.uids) {
+		t.Fatalf("uids don't match, expected '%v', got '%v'", expected.uids, got.uids)
```

Comment:
is it the number of uids that doesn't match, not the actual uids themselves?

Reply:
It's the number, but I figure the failed test output is more helpful if it shows both slices in full